core/src/main/java/com/mongodb/hadoop/util/MongoConfigUtil.java

```diff
@@ -1062,33 +1151,33 @@ public static void close(final Mongo client) {
                 client.close();
             }
     }
-    
+
     private static MongoClient getMongoClient(final MongoClientURI uri) throws UnknownHostException {
         MongoClient mongoClient = CLIENTS.get().get(uri);
-            if (mongoClient == null) {
-                if (uri.getOptions().isSslEnabled()) {
-                    MongoClientOptions options = MongoClientOptions.builder(uri.getOptions()).socketFactory(sslContext.getSocketFactory()).build();
-                    final List<ServerAddress> seedList;
-                    if (uri.getHosts().size() == 1) {
-                        seedList = Collections.singletonList(new ServerAddress(uri.getHosts().get(0)));
-                    } else {
-                        seedList = new ArrayList<ServerAddress>(uri.getHosts().size());
-                        for (final String host : uri.getHosts()) {
-                            seedList.add(new ServerAddress(host));
-                        }
-                    }
-                    List<MongoCredential> credentials = uri.getCredentials() != null ?
-                            Collections.singletonList(uri.getCredentials()) :
-                            Collections.<MongoCredential>emptyList();
-                    mongoClient =  new MongoClient(seedList, credentials, options);
+        if (mongoClient == null) {
+            if (uri.getOptions().isSslEnabled() && !uri.getURI().startsWith("mongodb+srv://")) {
```

Comment:
@mlsaito A [job](https://dashboard-staging.xplenty.com/krishna/jobs/1889122) with the [old code](https://github.com/xplenty/mongo-hadoop/blob/0b9f5684a7cb7bee47a448fd4c08d8a757293a1c/core/src/main/java/com/mongodb/hadoop/util/MongoConfigUtil.java#L1069) returned failed when the connection scheme is `srv`. So I put the new implementation to check `srv` on this block otherwise it will go to else block as [SRV is automatically set to be SSL](https://docs.mongodb.com/manual/reference/connection-string/). Any thoughts about it?

Reply:
@krishnapriawan This makes sense. Thanks!